### PR TITLE
added feature incremental sync for product categories

### PR DESCRIPTION
**Pull request by chronos25**

### CODE REVIEW

File: admin/src/app/home/connect/connect.component.html [hunk 1]
```diff
@@ -116,7 +116,7 @@
                 </button>
               </td>
             </tr>
-            <tr>
+            <!-- <tr>
```
Comment (with-shrey) [inline admin/src/app/home/connect/connect.component.html:119]: Remove commented code

File: admin/src/app/home/connect/connect.component.html [hunk 2]
```diff
@@ -255,6 +255,24 @@
...
+            <tr>
+              <td>
+                Product Categories
+              </td>
+              <td>
+                <label class="switch switch-3d switch-primary">
+                  <input type="checkbox" class="switch-input" [ngModel]="checkSync('product_types')" disabled>
+                  <span class="switch-slider"></span>
+                </label>
+                <div [ngClass]="{'d-none': !syncing.sales}" class="spinner-border spinner-border-sm spinner-margins"
```
Comment (with-shrey) [inline admin/src/app/home/connect/connect.component.html:267]: syncing.sales ?

File: admin/src/app/home/connect/connect.component.ts [hunk 3]
```diff
@@ -198,7 +198,7 @@ export class ConnectComponent implements OnInit, OnDestroy {
         });
   }
 
-  private syncVendProductTypes() {
+  /*private syncVendProductTypes() {
```
Comment (with-shrey) [inline admin/src/app/home/connect/connect.component.ts:201]: remove commented code

File: common/models/org-model.js [hunk 4]
```diff
@@ -934,7 +934,7 @@ module.exports = function (OrgModel) {
                 });
         };
 
-        OrgModel.remoteMethod('syncVendProductTypes', {
+       /* OrgModel.remoteMethod('syncVendProductTypes', {
```
Comment (with-shrey) [inline common/models/org-model.js:937]: commented code

File: common/models/sync-model.js [hunk 5]
```diff
@@ -672,14 +654,14 @@ module.exports = function (SyncModel) {
             });
     };
 
-    SyncModel.syncVendProductTypes = function (id, options) {
+  /*  SyncModel.syncVendProductTypes = function (id, options,versionsAfter = 0) {
```
Comment (with-shrey) [inline common/models/sync-model.js:657]: commented code

File: workers/syncWorker.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ var db = null;
 const _ = require('underscore');
 const SUCCESS = 0;
 const FAILURE = 1;
-const syncInterval = 300;
```
Comment (with-shrey) [inline workers/syncWorker.js:16]: revert this

File: workers/workers-v2/fetch-incremental-category/fetch-incremental-category.js [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    })
+    .finally(function(){
+        logger.debug({
+            orgModelId,
+            message: 'Closing DB Connection'
+        });
+        if(db){
+            return db.close();
+        }
+    }).catch(function(error){
```
Comment (with-shrey) [inline workers/workers-v2/fetch-incremental-category/fetch-incremental-category.js:37]: .catch on new line

File: workers/workers-v2/fetch-incremental-category/fetch-incremental-category.js [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+            .catch(function(error){
+                logger.error({
+                    message:'Could not fetch category from vend',
+                    orgModelId,
+                    categoryBatchNumber,
+                    error,
+                    functionName: 'fetchCategory'
+                })
+                return Promise.reject('Could not fetch category from vend');
+            }).then(function(response){
```
Comment (with-shrey) [inline workers/workers-v2/fetch-incremental-category/fetch-incremental-category.js:67]: new line before .then

**Review by with-shrey — CHANGES_REQUESTED** [review]

File: admin/src/app/home/connect/connect.component.html (hunk 1 above)
Comment (chronos25) [reply admin/src/app/home/connect/connect.component.html:119]: resolved !

File: admin/src/app/home/connect/connect.component.html (hunk 2 above)
Comment (chronos25) [reply admin/src/app/home/connect/connect.component.html:267]: resolved !

File: admin/src/app/home/connect/connect.component.ts (hunk 3 above)
Comment (chronos25) [reply admin/src/app/home/connect/connect.component.ts:201]: resolved !

File: common/models/sync-model.js (hunk 5 above)
Comment (chronos25) [reply common/models/sync-model.js:657]: Resolved !

File: workers/syncWorker.js (hunk 6 above)
Comment (chronos25) [reply workers/syncWorker.js:16]: Reverted !

**Comment by with-shrey** [comment]
LGTM

**Pull request merged**